src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesService.kt

```diff
@@ -45,6 +49,10 @@ class RolesService(
   fun updateRoleAdminType(roleCode: String, roleAmendment: RoleAdminTypeAmendment) {
     authService.updateRoleAdminType(roleCode, roleAmendment)
   }
+
+  private fun Set<AdminType>.isDPSRole(): Boolean = (DPS_ADM in this) or (DPS_LSA in this)
+  private fun List<AdminTypeReturn>.isDPSRole(): Boolean =
+    map { it.adminTypeCode }.any { (it == DPS_ADM.adminTypeCode || (it == DPS_LSA.adminTypeCode)) }
```

Comment:
We have 2 different AdminType objects - so it's kinda duplicated code to effectively do the same thing (any ideas to improve would be gratefully received)